File: AlternateIconsSwift/AlternateIconsSwift/ViewController.swift
//
//  ViewController.swift
//  AlternateIconsSwift
//
//  Created by Cesar Cavazos on 4/6/17.
//  Copyright © 2017 Propelics. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func changeAppIcon(iconName: String?) {
        if UIApplication.shared.supportsAlternateIcons {
            print("App Icon can be changed")
        } else{
            print("App Icon cannot be changed")
            return
        }
        
        UIApplication.shared.setAlternateIconName(iconName) { (err:Error?) in
            if let error = err {
                print("Error changing the icon ", error.localizedDescription)
            }
        }
    }
    
    @IBAction func handleAlternateIcon2(_ sender: Any) {
        changeAppIcon(iconName: "propelicsLightTheme")
    }
    
    @IBAction func handleAlternateIcon1(_ sender: Any) {
        changeAppIcon(iconName: "propelicsDarkTheme")
    }
    
    @IBAction func handleDefaultIcon(_ sender: Any) {
        changeAppIcon(iconName: nil)
    }
}

File: AlternateIconsXamarin/AlternateIconsXamarin/ViewController.cs
using System;

using UIKit;

namespace AlternateIconsXamarin
{
	public partial class ViewController : UIViewController
	{
		protected ViewController(IntPtr handle) : base(handle)
		{
			// Note: this .ctor should not contain any initialization logic.
		}

		public override void ViewDidLoad()
		{
			base.ViewDidLoad();
			// Perform any additional setup after loading the view, typically from a nib.
		}

		public override void DidReceiveMemoryWarning()
		{
			base.DidReceiveMemoryWarning();
			// Release any cached data, images, etc that aren't in use.
		}

		private void changeAppIcon(String iconName)
		{
			if (UIApplication.SharedApplication.SupportsAlternateIcons)
			{
				Console.WriteLine("App Icon can be changed");
			}
			else 
			{ 
				Console.WriteLine("App Icon cannot be changed");
				return;
			}

			UIApplication.SharedApplication.SetAlternateIconName(iconName, (err) =>
			{
				Console.WriteLine("Error changing the icon: {0}", err);
			});
		}

		partial void DefaultButton_TouchUpInside(UIButton sender)
		{
			changeAppIcon(null);
		}

		partial void DarkThemeButton_TouchUpInside(UIButton sender)
		{
            changeAppIcon("propelicsDarkTheme");
		}

		partial void LightThemeButton_TouchUpInside(UIButton sender)
		{
            changeAppIcon("propelicsLightTheme");
		}
	}
}
